src/server/products/delete-product.server.ts

```diff
@@ -0,0 +1,19 @@
+'use server'
+import { connectToDb } from '@/server/connectToDb'
+import { Product } from '@/server/products/productSchema'
+
+export async function deleteProduct(id?: string) {
+  'use server'
+  try {
+    await connectToDb()
+
+    const deletedProduct = await Product.findByIdAndDelete(id)
```

Comment:
you don't delete photos from cloudinary when deleting, see example in editing product

Reply:
fixed it 👍️